tests/tel2puml/main/test_main.py

```diff
@@ -0,0 +1,134 @@
+    """Tests for the function find_json_files"""
+    nested_dir_1 = tmp_path / "dir1"
+    nested_dir_1.mkdir()
+
+    nested_dir_2 = nested_dir_1 / "dir2"
+    nested_dir_2.mkdir()
+
+    json_file_1 = nested_dir_1 / "file1.json"
+    json_file_1.write_text("{}")
+
+    json_file_2 = nested_dir_2 / "file2.json"
+    json_file_2.write_text("{}")
+
+    txt_file = nested_dir_1 / "file.txt"
+    txt_file.write_text("This is a text file")
+
+    result = find_json_files(str(tmp_path))
+
+    assert sorted(result) == sorted([str(json_file_1), str(json_file_2)])
+
+
+def test_generate_components_options(
+    mock_yaml_config_file: Path, tmp_path: Path
+) -> None:
+    """Tests the function generate_component_options"""
+
+    # Test 1: otel2puml command
+    command: Literal["otel2puml", "otel2pv", "pv2puml"] = "otel2puml"
+    args_dict = {
+        "command": "otel2puml",
+        "output_file_directory": ".",
+        "config_file": str(mock_yaml_config_file),
+        "ingest_data": True,
+        "find_unique_graphs": False,
+    }
+    otel_pv_options, pv_puml_options = generate_component_options(
+        command, args_dict
+    )
+    assert otel_pv_options
+    assert otel_pv_options["config"]
+    assert otel_pv_options["ingest_data"]
+    assert not pv_puml_options
+
+    # Test 2: otel2pv command
+    command = "otel2pv"
+    args_dict = {
+        "command": "otel2pv",
+        "output_file_directory": ".",
+        "config_file": str(mock_yaml_config_file),
+        "ingest_data": False,
+        "find_unique_graphs": False,
+    }
+    otel_pv_options, pv_puml_options = generate_component_options(
+        command, args_dict
+    )
+    assert otel_pv_options
+    assert otel_pv_options["config"]
+    assert not otel_pv_options["ingest_data"]
+    assert not pv_puml_options
+
+    # Test 3: pv2puml command, file_paths provided
+    command = "pv2puml"
+    with tempfile.NamedTemporaryFile(suffix=".json") as tmp_file:
+        args_dict = {
+            "command": "pv2puml",
+            "output_file_directory": ".",
+            "folder_path": None,
+            "file_paths": [tmp_file.name],
+            "job_name": "job_001",
+            "group_by_job": False,
+        }
+        otel_pv_options, pv_puml_options = generate_component_options(
+            command, args_dict
+        )
+        assert not otel_pv_options
+        assert pv_puml_options
+        assert pv_puml_options["file_list"] == [tmp_file.name]
+        assert pv_puml_options["job_name"] == "job_001"
+        assert not pv_puml_options["group_by_job_id"]
+
+    # Test 4: pv2puml command, directory provided
+    nested_dir_1 = tmp_path / "dir1"
+    nested_dir_1.mkdir()
+
+    json_file_1 = nested_dir_1 / "file1.json"
+    json_file_1.write_text("{}")
+    args_dict = {
+        "command": "pv2puml",
+        "output_file_directory": ".",
+        "folder_path": nested_dir_1,
+        "file_paths": None,
```

Comment:
Can yo check the case of an empty list here as well (thats what will be provided by the cli)